### DISCUSSION TITLE
Support for building pure wheel using scikit-build-core

### DISCUSSION THREAD

**Discussion by jcfr** [scikit-build/scikit-build-core#1463] [Ideas]
To support using CMake as scripting language at wheel build time (e.g for downloading files), I suggest we support declaring the wheel as pure.

This would allow me to address https://github.com/ImagingDataCommons/idc-index/issues/41 where we are working on downloading and packing files that we would prefer not to add to the Git history.

**Comment by henryiii** [comment]
We do. `wheel.platlib = False` IIRC.

**Reply by jcfr** [reply]
I indeed tried this approach without success, the platform specific tag is still appended.

In this particular case, I get a wheel named:

```
idc_index-0.3.2.post1.dev8+g9d00435.d20240227-py3-none-manylinux_2_31_x86_64.whl
```

whereas I would have expected to have

```
idc_index-0.3.2.post1.dev8+g9d00435.d20240227-py3-none-any.whl
```

For reference, here is what I tried:

```diff
diff --git a/pyproject.toml b/pyproject.toml
index 2f3380d..de52650 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -63,14 +63,13 @@ Discussions = "https://github.com/ImagingDataCommons/idc-index/discussions"
 Changelog = "https://github.com/ImagingDataCommons/idc-index/releases"
 
 
-
 [tool.scikit-build]
-minimum-version = "0.4"
+minimum-version = "0.8"
 build-dir = "build/{wheel_tag}"
 metadata.version.provider = "scikit_build_core.metadata.setuptools_scm"
 sdist.include = ["idc_index/_version.py"]
+wheel.platlib = false
 wheel.py-api = "py3"
-wheel.expand-macos-universal-tags = true
```


![image](https://github.com/scikit-build/scikit-build/assets/219043/1cbeeb24-eb5e-4c68-9026-7de0e45e48ff)

**Reply by jcfr** [reply]
Looks like we would need to improve both `WheelTag.compute_best`[^1] to set skip `best_tag.platform` and use `any` if building a pure wheel.

[^1]: https://github.com/scikit-build/scikit-build-core/blob/57e4ac91f766f426187748546a4956ef466914c6/src/scikit_build_core/builder/wheel_tag.py#L34-L103

**Reply by jcfr** [reply]
See https://github.com/scikit-build/scikit-build-core/pull/655
